chandra_aca/dark_subtract.py

```diff
@@ -0,0 +1,431 @@
+    time : CxoTimeLike
+        Reference time for the dark calibration image.
+
+    Returns
+    -------
+    dc_img : np.array
+        Dark calibration image.
+    dc_tccd : float
+        Dark calibration CCD temperature.
+    """
+    dc = get_dark_cal_props(time, "nearest", include_image=True, aca_image=False)
+    dc_img = dc["image"]
+    if "t_ccd" in dc:
+        dc_tccd = dc["t_ccd"]
+    elif "ccd_temp" in dc:
+        dc_tccd = dc["ccd_temp"]
+    else:
+        raise ValueError("Dark cal must have t_ccd or ccd_temp")
+    return dc_img, dc_tccd
+
+
+@retry.retry(exceptions=requests.exceptions.RequestException, delay=5, tries=3)
+def get_tccd_data(start: CxoTimeLike, stop: CxoTimeLike, source="maude"):
+    """
+    Get the CCD temperature for a given time range.
+
+    This is a light wrapper around cheta.fetch_sci.Msid("aacccdpt", start, stop).
+
+    Parameters
+    ----------
+    start : CxoTimeLike
+        Start time of the time range.
+    stop : CxoTimeLike
+        Stop time of the time range.
+    source : str, optional
+        Source of the CCD temperature data. If 'maude', override the fetch_sci data source
+        to 'maude allow_subset=False'. Else, use the cheta default.
+
+    Returns
+    -------
+    t_ccd : np.array
+        CCD temperature values.
+    t_ccd_times : np.array
+        Times corresponding to the CCD temperature values.
+    """
+    if source == "maude":
+        # Override the data_source to be explicit about maude source.
+        with fetch_sci.data_source("maude allow_subset=False"):
+            dat = fetch_sci.Msid("aacccdpt", start, stop)
+    else:
+        dat = fetch_sci.Msid("aacccdpt", start, stop)
+    t_ccd = dat.vals
+    t_ccd_times = dat.times
+    return t_ccd, t_ccd_times
+
+
+@retry.retry(exceptions=requests.exceptions.RequestException, delay=5, tries=3)
+def get_dcsub_aca_images(
+    start=None,
+    stop=None,
+    aca_images=None,
+    t_ccd=None,
+    t_ccd_times=None,
+    dc_img=None,
+    dc_tccd=None,
+    source="maude",
+    full_table=False,
+):
+    """
+    Get dark current subtracted ACA images for a given time range.
+
+    Parameters
+    ----------
+    start : str, optional
+        Start time of the time range. Default is None.
+    stop : str, optional
+        Stop time of the time range. Default is None.
+    aca_images : dict, optional
+        Dictionary of ACA image tables. Should be keyed by slot. Default is None.
+        This is intended for testing.
+    t_ccd : array-like, optional
+        Array of CCD temperatures. Default is None.
+    t_ccd_times : array-like, optional
+        Array of CCD temperature times. Default is None.
+    dc_img : array-like, optional
+        Dark current image. Default is None.
+    dc_tccd : float, optional
+        Dark current CCD temperature. Default is None.
+    source : str, optional
+        Source of the ACA images. Can be 'maude' or 'mica'. Default is 'maude'.
+    full_table : bool, optional
```

Comment:
We can probably cut the full_table option and just always have this method return the full table.